src/main/java/org/jitsi/videobridge/util/JvbConfig.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.util;
+
+import com.typesafe.config.*;
+import org.jetbrains.annotations.*;
+import org.jitsi.utils.logging2.*;
+
+import java.nio.file.*;
+import java.util.function.*;
+
+public class JvbConfig
+{
+    protected static Logger logger = new LoggerImpl(JvbConfig.class.getName());
+
+    protected static Config config;
+    protected static Config legacyConfig;
+    protected static String commandLineArgs;
+
+    public static Supplier<Config> configSupplier = ConfigFactory::load;
+    public static Supplier<Config> legacyConfigSupplier = () -> {
+        String oldConfigHomeDirLocation = System.getProperty("net.java.sip.communicator.SC_HOME_DIR_LOCATION");
+        String oldConfigHomeDirName = System.getProperty("net.java.sip.communicator.SC_HOME_DIR_NAME");
+        try
+        {
+            return ConfigFactory.parseFile(
+                    Paths.get(oldConfigHomeDirLocation, oldConfigHomeDirName, "sip-communicator.properties")
+                            .toFile());
+        }
+        catch (InvalidPathException | NullPointerException e)
+        {
+            logger.info("No legacy config file found");
+            return ConfigFactory.parseString("");
+        }
+    };
+    public static Supplier<String> commandLineArgsSupplier = () -> config.getString("sun.java.command");
```

Comment:
good catch.  yeah i think passing them to init seems like it'd work, i'll do that.

Reply:
> 2) it loses argument segmentation (i.e. java foo "Hello world" and java foo Hello world appear the same.

with regards to this, i agree it's totally naive.  the command line args are deprecated with new config anyway (an overrides file is passed, or, they're defined as system properties) so i only worried about handling the args we previously had and none made sense with spaces.  but, with your recommendation for 1 this'll be moot anyways i guess.

Reply:
another idea on the command-line args thing:  what about saving the program arguments into a custom system property and access them via that (instead of `sun.java.command`)?  we do run into the splitting problem a bit, but: 1) like i said, the arguments we supported are limited and spaces aren't valid in any of them 2) we could use a custom delimiter between the args and split on that instead of space which would make things a bit more robust?

i looked at passing them to `init`, but then `reloadConfig` has to re-pass them in, which is a little weird.  i then looked at splitting the config file logic out of `init` into a `loadConfigFiles` helper, like this:

```
public static void init(String[] programArgs)
    {
        commandLineArgs = programArgs;
        loadConfigFiles();
    }

   protected static void loadConfigFiles()
   {
       config = configSupplier.get();
       legacyConfig = legacyConfigSupplier.get();
       <snip>
    }

    public static void reloadConfig()
    {
        logger.info("Reloading config");
        ConfigFactory.invalidateCaches();
        loadConfigFiles();
    }
````
but then unit tests can no longer alter the configs by just calling `reloadConfig`, they'd have to re-call init each time--not the worst thing but it feels a little off, semantically.  it seems like this should able to work via `reloadConfig`

just some thoughts

Reply:
Making unit tests jump through some hoops to change `programArgs` doesn't seem so bad to me -- this isn't something anything would normally ever want to do outside a unit test.

Maybe a `setProgramArgs` method that the unit tests (and a program potentially doing something weird) can use?

Reply:
ok, i took another shot at cleaning that up.